### PR TITLE
Restructuring tools and making some APIs public

### PR DESCRIPTION
**Pull request by nsivabalan**
1. DumpData has grown to take in lot many operations and it was hard to maintain it. So, have restructured into many classes viz DumpLog, DumpIndex, DumpReplicaToken and DumpData. From now on, DumpDataHelper will be the only class which deals with reading data from disk. Rest of the classes acts as a wrapper in performing the operation. 
2. Introduced config files for each classes (DumpLogConfig, DumpIndexConfig, DumpReplicaTokenConfig and DumpReplicaTokenConfig)
3. All these tools take in a property file as input and reads all the property from the same.
4. Made some of these operations as public.

SLA: 20 mins
Reviewers: @vgkholla

### CODE REVIEW

**Comment by nsivabalan** [comment]
synced up offline. Addressed all the comments. 
@vgkholla : you can go ahead and review it

File: ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java [hunk 1]
```diff
@@ -39,79 +34,59 @@
  * 2) In the index file boundaries on all replicas in a partition
  */
 public class ConsistencyCheckerTool {
+  private final VerifiableProperties verifiableProperties;
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java]: any reason for this to be a class variable? Might have missed it but is it used anywhere else other than the constructor?

File: ambry-tools/src/main/java/com.github.ambry/store/DumpDataTool.java [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+public class DumpDataTool {
+  private final ClusterMap clusterMap;
+  // The index file that needs to be dumped for comparison purposes
+  @Config("file.to.read")
+  private final String fileToRead;
+
+  /**
+   * File path referring to the hardware layout
+   */
+  @Config("hardware.layout.file.path")
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/store/DumpDataTool.java]: not sure if config decorations make sense for private variables (here and in consistency checker)

File: ambry-tools/src/main/java/com.github.ambry/store/DumpDataTool.java [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+    clusterMap = new ClusterMapManager(hardwareLayoutFilePath, partitionLayoutFilePath,
+        new ClusterMapConfig(new VerifiableProperties(new Properties())));
+  }
+
+  public static void main(String args[]) {
+    try {
+      VerifiableProperties verifiableProperties = StoreToolsUtil.getVerifiableProperties(args);
+      DumpDataTool dumpDataTool = new DumpDataTool(verifiableProperties);
+      dumpDataTool.doOperation();
+    } catch (Exception e) {
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/store/DumpDataTool.java]: why does the exception have to be caught?

**Comment by vgkholla** [comment]
Will import it in Intellij and give it another look tomorrow.

File: ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java (hunk 1 above)
Comment (nsivabalan) [inline ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java]: Yes, the dumpIndexTool needs the verifiable props to generate the config by itself.

File: ambry-tools/src/main/java/com.github.ambry/store/DumpDataTool.java (hunk 2 above)
Comment (nsivabalan) [inline ambry-tools/src/main/java/com.github.ambry/store/DumpDataTool.java]: have fixed it

**Comment by nsivabalan** [comment]
I have addressed your comments so far. Will wait for you review.

File: ambry-tools/src/main/java/com.github.ambry/store/DumpReplicaTokenTool.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    clusterMap = new ClusterMapManager(hardwareLayoutFilePath, partitionLayoutFilePath,
+        new ClusterMapConfig(new VerifiableProperties(new Properties())));
+  }
+
+  public static void main(String args[]) {
+    try {
+      VerifiableProperties verifiableProperties = StoreToolsUtil.getVerifiableProperties(args);
+      DumpReplicaTokenTool dumpReplicaTokenTool = new DumpReplicaTokenTool(verifiableProperties);
+      dumpReplicaTokenTool.dumpReplicaToken();
+    } catch (Exception e) {
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/store/DumpReplicaTokenTool.java]: exception doesn't need to be caught.

File: ambry-tools/src/main/java/com.github.ambry/store/DumpLogTool.java [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+    }
+  }
+
+  public static void main(String args[]) {
+    try {
+      VerifiableProperties verifiableProperties = StoreToolsUtil.getVerifiableProperties(args);
+      DumpLogTool dumpLogTool = new DumpLogTool(verifiableProperties);
+      dumpLogTool.doOperation();
+    } catch (Exception e) {
+      logger.error("Closed with exception ", e);
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/store/DumpLogTool.java]: exception doesn't need to be caught

File: ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java [hunk 6]
```diff
@@ -39,79 +32,45 @@
...
-          Boolean.parseBoolean(options.valueOf(includeAcceptableInconsistentBlobsOpt));
-
-      ConsistencyCheckerTool consistencyCheckerTool =
-          new ConsistencyCheckerTool(map, includeAcceptableInconsistentBlobs);
-      consistencyCheckerTool.checkConsistency(rootDirectoryForPartition);
+      VerifiableProperties verifiableProperties = StoreToolsUtil.getVerifiableProperties(args);
+      ConsistencyCheckerTool consistencyCheckerTool = new ConsistencyCheckerTool(verifiableProperties);
+      consistencyCheckerTool.checkConsistency();
     } catch (Exception e) {
       System.err.println("Consistency checker exited with Exception: " + e);
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java]: exception does not need to be caught. Also, not using `logger`.

File: ambry-tools/src/main/java/com.github.ambry/store/DumpIndexTool.java [hunk 7]
```diff
@@ -0,0 +1,497 @@
...
+        new ClusterMapConfig(verifiableProperties));
+  }
+
+  public static void main(String args[]) {
+    try {
+      VerifiableProperties verifiableProperties = StoreToolsUtil.getVerifiableProperties(args);
+      DumpIndexTool dumpIndexTool = new DumpIndexTool(verifiableProperties);
+      dumpIndexTool.doOperation();
+    } catch (Exception e) {
+      logger.error("Closed with exception ", e);
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/store/DumpIndexTool.java]: exception does not need to be caught.

**Comment by nsivabalan** [comment]
Yes, built using java 7 and formatted as well.

**Pull request merged**